src/connect.ts

```diff
@@ -288,26 +289,28 @@ export async function connectMongoClient(
     state: DevtoolsConnectionState
   }> {
   detectAndLogMissingOptionalDependencies(logger);
+  // If PROVIDER_NAME was specified to the MongoClient options, adding callbacks would conflict
+  // with that; we should omit them so that e.g. mongosh users can leverage the non-human OIDC
+  // auth flows by specifying PROVIDER_NAME.
+  const shouldAddOidcCallbacks = isHumanOidcFlow(uri, clientOptions);
+  const state = clientOptions.parentState ?? new DevtoolsConnectionState(clientOptions, logger);
+  const mongoClientOptions: MongoClientOptions & Partial<DevtoolsConnectOptions> =
+    merge({}, clientOptions, shouldAddOidcCallbacks ? state.oidcPlugin.mongoClientOptions : {});
+
+  mongoClientOptions.lookup = (hostname, options, callback) => {
+    return dns.lookup(hostname, { ...options, verbatim: false }, callback);
```

Comment:
This could even be a default only, right, in case the Node.js driver e.g. starts passing its own `verbatim` option?

```suggestion
    return dns.lookup(hostname, { verbatim: false, ...options }, callback);
```

Also, I would add a comment here that explains why we are adding a custom `lookup` option.